sdk/resourcemanager/Azure.ResourceManager.Core/src/ApiVersions.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.ResourceManager.Resources;
+using System.Reflection;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A class representing Azure resource manager client options.
+    /// </summary>
+    public class ApiVersions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApiVersions"/> class.
+        /// </summary>
+        public ApiVersions()
+        {
+            BuildApiTable();
+        }
+
+        private Dictionary<string, string> _resourceToApiVersions = new Dictionary<string, string>();
+
+        private void BuildApiTable()
+        {
+            var methods = GetExtensionMethods();
+            foreach (var method in methods)
+            {
+                if (method.Name.EndsWith("RestApiVersions"))
+                {
+                    var apiObject = method.Invoke(null, new object[] { this });
+                    var properties = apiObject.GetType().GetProperties();
+                    foreach (var prop in properties)
+                    {
+                        if (typeof(ApiVersionsBase).IsAssignableFrom(prop.PropertyType))
+                        {
+                            var propVal = (ApiVersionsBase)prop.GetValue(apiObject);
+                            Console.WriteLine(prop.GetType());
+                            var key = propVal.ResourceType;
+                            _resourceToApiVersions.Add(key.ToString(), propVal);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static IEnumerable<MethodInfo> GetExtensionMethods()
+        {
+            var results =
```

Comment:
is there a way to just get all extensions from this.GetType()? instead of needing to sort through all types in the current appdomain?

Reply:
Have not found a easier way to do this. What are you thinking here?

Reply:
I guess what I was after doesn't exist here